### PR TITLE
Terrain Control Button

### PR DESCRIPTION
**Pull request by acalcutt**
This is my code to add a Terrain button control to maplbre like mentioned in ( https://github.com/prozessor13/maplibre-gl-js/issues/37 )

This code adds a clickable button control to toggle terrain. 
terrain-toggle

The toggle button itself is executing addTerrain, so the options for addTerrain get specified  as part of the button options like this
```
  map.addControl(
    new maplibregl.TerrainControl({
      id: "terrain"
      options: {
        exaggeration: 2,
        elevationOffset: 0
      }
    })
  );
```

The one change I made to the code outside of this button was to change the type of exaggeration from boolean to number, because it seemed like it is used as a multiplier

map.ts
`addTerrain(id: string, options?: {**exaggeration: number**; elevationOffset: number}) {`

terrain_source_cache.ts 
`enable(sourceCache: SourceCache, options?: {**exaggeration: number**; elevationOffset: number}): void {`

- [X] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [X] Briefly describe the changes in this PR.
 - [X] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality.
 - [ ] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [ ] Manually test the debug page.
 - [ ] Suggest a changelog category: bug/feature/docs/etc. or "skip changelog".
 - [ ] Add an entry inside this element for inclusion in the `maplibre-gl-js` changelog: `<changelog></changelog>`.

### CODE REVIEW

**Review by prozessor13 — APPROVED** [review]

**Comment by acalcutt** [comment]
Thanks for including this. sorry my commit history is such a mess with all those merges... lol

**Comment by prozessor13** [comment]
No Problem, this is the way of git :), Thx for your pull-request!

File: debug/terrain_control.html [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+map.on("load", () => {
+  map.addSource("terrain", {
+     "type": "raster-dem",
+     "tiles": ["https://vtc-cdn.maptoolkit.net/terrainrgb/{z}/{x}/{y}.webp"],
+     "encoding": "mtk",
+     "maxzoom": 14
+  });
+  map.addControl(
+    new maplibregl.TerrainControl({
+      id: "terrain"
```
Comment (acalcutt) [inline debug/terrain_control.html:39]: This line is missing a comma, it should be 
`      id: "terrain",`

Comment (acalcutt) [reply debug/terrain_control.html:39]: @prozessor13 just wanted to mentions this. I noticed there is a missing comma in my example that was merged in.

**Pull request merged**